core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2665,8 +2665,17 @@ object SparkContext extends Logging {
     // When running locally, don't try to re-execute tasks on failure.
     val MAX_LOCAL_TASK_FAILURES = 1
 
+    val cpusPerTask = sc.conf.get(CPUS_PER_TASK)
+
+    def checkClusterExecutorCores(): Unit = if (sc.conf.get(EXECUTOR_CORES) < cpusPerTask) {
+      throw new SparkException(s"${EXECUTOR_CORES.key} must not be < ${CPUS_PER_TASK.key}.")
+    }
+
     master match {
       case "local" =>
+        if (cpusPerTask > 1) {
```

Comment:
(Why not?)

Reply:
Because when run local mode, just 1 core is available.
```$SPARK_HOME/bin/spark-shell --master local --conf spark.task.cpus=2
scala>sc.setLogLevel("INFO")
scala>sc.parallelize(1 to 9).collect```
You can see spark will hang after log `INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks`.

Reply:
I see, local == local[1]